### DISCUSSION TITLE
Is it possible to switch validations based on the events?

### DISCUSSION THREAD

**Discussion by manak1** [react-hook-form/react-hook-form#4700] [Q&A] [answered]
Hello, I have a question about switching the validations based on the events.

For example on the onChange event I want to check if the email doesn't contain any characters except  alphanumeric. 
On the `onBlur` event I want to check if the email is valid.

Is this kind future supported on react-hook-form?

Thank you for reading my question.

**Comment by bluebill1049** [comment, marked as answer]
we don't have such mechasiam.

**Reply by manak1** [reply]
Thank you for answering my question! :D
